components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/util/ActionManagementAuditLoggerTest.java

```diff
@@ -0,0 +1,178 @@
+    private MockedStatic<IdentityUtil> identityUtil;
+    private MockedStatic<IdentityTenantUtil> identityTenantUtil;
+    private MockedStatic<LoggerUtils> loggerUtilsMockedStatic;
+
+    @BeforeMethod
+    public void setUp() throws NoSuchFieldException, IllegalAccessException {
+
+        MockitoAnnotations.openMocks(this);
+        auditLogger = new ActionManagementAuditLogger();
+        identityUtil = mockStatic(IdentityUtil.class);
+        identityTenantUtil = mockStatic(IdentityTenantUtil.class);
+
+        carbonContextMockedStatic = mockStatic(CarbonContext.class);
+        carbonContext = mock(CarbonContext.class);
+        carbonContextMockedStatic.when(CarbonContext::getThreadLocalCarbonContext).thenReturn(carbonContext);
+        when(carbonContext.getUsername()).thenReturn("testUser");
+        when(carbonContext.getTenantDomain()).thenReturn("carbon.super");
+        identityUtil.when(() -> IdentityUtil.getInitiatorId("testUser", "carbon.super")).
+                thenReturn("initiator-id-test");
+
+        loggerUtilsMockedStatic = mockStatic(LoggerUtils.class);
+        loggerUtilsMockedStatic.when(LoggerUtils::isEnableV2AuditLogs).thenReturn(true);
+        loggerUtilsMockedStatic.when(() -> LoggerUtils.jsonObjectToMap(any(JSONObject.class))).thenCallRealMethod();
+
+        // Mock Action
+        action = mock(Action.class);
+        when(action.getId()).thenReturn("action-test-id");
+        when(action.getName()).thenReturn("Test Action");
+        when(action.getDescription()).thenReturn("This is a test action.");
+        when(action.getType()).thenReturn(Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN);
+        Map<String, String> authProperties = new HashMap<>();
+        authProperties.put("accessToken", "W*********t");
+        Authentication auth = new Authentication.AuthenticationBuilder().type(
+                Authentication.Type.BEARER).properties(authProperties).build();
+        when(action.getEndpoint()).thenReturn(new EndpointConfig.EndpointConfigBuilder().
+                uri("https://test.com").
+                authentication(auth).build());
+        when(action.getStatus()).thenReturn(Action.Status.ACTIVE);
+    }
+
+    @AfterMethod
+    public void tearDown() {
+
+        auditLogger = null;
+        action = null;
+        carbonContextMockedStatic.close();
+        identityUtil.close();
+        identityTenantUtil.close();
+        loggerUtilsMockedStatic.close();
+    }
+
+    @Test
+    public void testPrintAuditLogWithAction() throws NoSuchFieldException, IllegalAccessException {
+
+        ActionManagementAuditLogger.Operation operation = ActionManagementAuditLogger.Operation.ADD;
+        auditLogger.printAuditLog(operation, action);
+        AuditLog.AuditLogBuilder capturedArg = captureTriggerAuditLogEventArgs();
+
+        Assert.assertNotNull(capturedArg);
+        Assert.assertEquals(extractDataField("ActionName", capturedArg), "Test Action");
+    }
```

Comment:
1. Looks like through out you have captured only specific fields. This returns a json right. That json needs to be validated.
2. Also IMO we should verify the audit log builder passed to triggerAuditLogEvent

This applies for all methods below